packages/live-status-gateway/src/collections/adLibActions.ts

```diff
@@ -0,0 +1,72 @@
+import { Logger } from 'winston'
+import { CoreHandler } from '../coreHandler'
+import { CollectionBase, Collection, CollectionObserver } from '../wsHandler'
+import { CoreConnection } from '@sofie-automation/server-core-integration'
+import { AdLibAction } from '@sofie-automation/corelib/dist/dataModel/AdlibAction'
+import { DBPartInstance } from '@sofie-automation/corelib/dist/dataModel/PartInstance'
+import { unprotectString } from '@sofie-automation/shared-lib/dist/lib/protectedString'
+import { PartInstanceName } from './partInstances'
+
+export class AdLibActionsHandler
+	extends CollectionBase<AdLibAction[]>
+	implements Collection<AdLibAction[]>, CollectionObserver<Map<PartInstanceName, DBPartInstance | undefined>>
+{
+	public observerName: string
+	private _core: CoreConnection
+	private _curRundownId: string | undefined
+
+	constructor(logger: Logger, coreHandler: CoreHandler) {
+		super('AdLibActionHandler', 'adLibActions', logger, coreHandler)
+		this._core = coreHandler.coreConnection
+		this.observerName = this._name
+	}
+
+	async changed(id: string, changeType: string): Promise<void> {
+		this._logger.info(`${this._name} ${changeType} ${id}`)
+		if (!this._collection) return
+		const col = this._core.getCollection<AdLibAction>(this._collection)
+		if (!col) throw new Error(`collection '${this._collection}' not found!`)
+		this._collectionData = col.find({ rundownId: this._curRundownId })
+		await this.notify(this._collectionData)
+	}
+
+	async update(source: string, data: Map<PartInstanceName, DBPartInstance | undefined> | undefined): Promise<void> {
+		this._logger.info(`${this._name} received partInstances update from ${source}`)
+		const prevRundownId = this._curRundownId
+		this._curRundownId = data ? unprotectString(data.get(PartInstanceName.current)?.rundownId) : undefined
+
+		await new Promise(process.nextTick.bind(this))
+		if (!this._collection) return
+		if (prevRundownId !== this._curRundownId) {
+			if (this._subscriptionId) this._coreHandler.unsubscribe(this._subscriptionId)
+			if (this._dbObserver) this._dbObserver.stop()
+			if (this._curRundownId) {
+				this._subscriptionId = await this._coreHandler.setupSubscription(this._collection, {
```

Comment:
I don't think the Publication name and the Collection name are strictly related. I think, by convention we've set up this so that there's a Publication by the name of the Collection, but I'm not sure if this should be reused like that.

Reply:
I've moved to using `CollectionName` and having separate `publication` and `collection` for their separate uses. It would be nice to have publication names exposed in a similar way to `CollectionName` (unless I've missed where they're exposed?) but I think exposing them is outside the scope of this PR because it touches a lot of Sofie code
 https://github.com/nrkno/sofie-core/pull/777/commits/6f0742863b0f042fc9f30caea2534bb7fb08d40c